sse.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022 Flamego. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package sse
+
+import (
+	"encoding/json"
+	"net/http"
+	"reflect"
+	"time"
+
+	"github.com/flamego/flamego"
+)
+
+const (
+	defaultPingInterval = 10 * time.Second
+)
+
+type Options struct {
+	// PingInterval is the time to wait between sending pings to the client.
+	PingInterval time.Duration
+}
+
+type Connection struct {
```

Comment:
Does this type need to be public? If so we need docstring, if not let's make it private.

Reply:
Hmmm, it should be private I think.